graphql/appointment/fields.ts

```diff
@@ -119,5 +117,33 @@ export class ExtendedFieldsLectureResolver {
         throw new Error('Cannot determine total of loose appointment');
     }
 
-    // TODO add displayName FieldResolver
+    @FieldResolver((returns) => String)
+    @Authorized(Role.USER)
+    async displayName(@Ctx() context: GraphQLContext, @Root() appointment: Appointment): Promise<string> {
+        switch (appointment.appointmentType) {
+            case lecture_appointmenttype_enum.match: {
+                let isOrganizer;
+
+                if (!isElevated(context) && !isSessionStudent(context)) {
+                    isOrganizer = false;
+                } else {
+                    isOrganizer = (await prisma.lecture.count({ where: { id: appointment.id, organizerIds: { has: context.user.userID } } })) > 0;
+                }
+
+                if (isOrganizer) {
+                    const [tutee] = await getUsers(appointment.participantIds);
+                    return `${tutee.firstname} ${tutee.lastname}`;
+                } else {
+                    const [tutor] = await getUsers(appointment.organizerIds);
```

Comment:
Do match appointments always only have one organizer and one participant?

Reply:
Yes, this may change in the future, but for now 1on1 learning support have one organizer and one participant.